lib/core/payjoin/data/datasources/pdk_payjoin_datasource.dart

```diff
@@ -77,25 +77,43 @@ class PdkPayjoinDatasource {
     required int expireAfterSec,
   }) async {
     try {
-      final payjoinDirectory = await Url.fromStr(_payjoinDirectoryUrl);
-
       final (ohttpKeys, ohttpRelay) = await fetchOhttpKeyAndRelay(
-        payjoinDirectory: payjoinDirectory,
+        payjoinDirectory: _payjoinDirectoryUrl,
       );
 
       if (ohttpRelay == null || ohttpKeys == null) {
         throw Exception('All OHTTP relays failed');
       }
 
-      final receiver = await Receiver.create(
+      final newReceiver = NewReceiver.create(
         address: address,
         network: isTestnet ? Network.testnet : Network.bitcoin,
-        directory: payjoinDirectory,
+        directory: _payjoinDirectoryUrl,
         ohttpKeys: ohttpKeys,
-        ohttpRelay: ohttpRelay,
         expireAfter: BigInt.from(expireAfterSec),
       );
 
+      final completer = Completer<ReceiverToken>();
+      final imp = InMemoryReceiverPersister();
+      final noOpPersister = DartReceiverPersister(
+        save: (receiver) async {
+          debugPrint('SAVING RECEIVER');
+          final token = await imp.save(receiver: receiver);
+          completer.complete(token);
+          return token;
+        },
+        load: (token) async {
+          final receiver = await imp.load(token: token);
+          return receiver;
+        },
+      );
+      final token = await newReceiver.persist(persister: noOpPersister);
+
+      final receiver = await Receiver.load(
+        token: token,
+        persister: noOpPersister,
+      );
+
```

Comment:
Is the completer necessary here? Can we use this more succinct approach like with the Sender below? (it compiles)

```suggestion
      final imp = InMemoryReceiverPersister();
      final persister = DartReceiverPersister(
        save: (receiver) async {
          return await imp.save(receiver: receiver);
        },
        load: (token) async {
          return await imp.load(token: token);
        },
      );
      final token = await newReceiver.persist(persister: persister);
      final receiver = await Receiver.load(
        token: token,
        persister: persister,
      );

```